### PR TITLE
schema: activate grub-dpkg deprecations

### PR DESCRIPTION
**Pull request by aciba90**
## Proposed Commit Message

```
schema: activate grub-dpkg deprecations
```

## Additional Context

SC-1173

## Checklist:

 - [x] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/topics/contributing.html)
 - [x] I have updated or added any unit tests accordingly
 - [x] I have updated or added any documentation accordingly

### CODE REVIEW

**Comment by aciba90** [comment]
<details>
  <summary>Generated doc</summary>
  image
</details>

File: cloudinit/config/schemas/schema-cloud-config-v1.json [hunk 1]
```diff
@@ -791,14 +791,24 @@
               "description": "Device to use as target for grub installation. If unspecified, ``grub-probe`` of ``/boot`` will be used to find the device"
             },
             "grub-pc/install_devices_empty": {
-              "type": ["string", "boolean"],
-              "description": "Sets values for ``grub-pc/install_devices_empty``. If unspecified, will be set to ``true`` if ``grub-pc/install_devices`` is empty, otherwise ``false``. Using a non-boolean value for this field is deprecated."
+              "description": "Sets values for ``grub-pc/install_devices_empty``. If unspecified, will be set to ``true`` if ``grub-pc/install_devices`` is empty, otherwise ``false``. Using a non-boolean value for this field is deprecated.",
+              "oneOf": [
+                {
+                  "type": "boolean"
```
Comment (blackboxsw) [inline cloudinit/config/schemas/schema-cloud-config-v1.json:797]: Added benefit of this ordering is that the docs will now list `(boolean/string)` which lends itself to listing the non-deprecated value first.

**Review by blackboxsw — APPROVED** [review]
LGTM. Validated behavior and expected messaging with:
```python
cat > invalid_dpkg.yaml <<EOF
grub_dpkg:                                                          
              enabled: true
              grub-pc/install_devices: /dev/sda                                 
              grub-pc/install_devices_empty: "false"
EOF
PYTHONPATH=. python3 -m cloudinit.cmd.main schema -c invalid_dpkg.yaml
Cloud config schema deprecations: grub_dpkg.grub-pc/install_devices_empty: DEPRECATED. Use a boolean value instead.
Valid cloud-config: invalid_dpkg.yaml


$ PYTHONPATH=. python3 -m cloudinit.cmd.main schema --docs cc_grub_dpkg # to compare this branch to main
```

File: cloudinit/config/schemas/schema-cloud-config-v1.json (hunk 1 above)
Comment (aciba90) [reply cloudinit/config/schemas/schema-cloud-config-v1.json:797]: That was intended. Thanks for noticing it!

**Pull request merged**